provisioning/service/tests/Config/IndividualEnrollmentTests.cs

```diff
@@ -44,8 +124,179 @@ public void IndividualEnrollment_Constructor_ThrowsOnInvalidParameters()
+            "                       \"issuerName\": \"CN=ROOT_00000000-0000-0000-0000-000000000000, OU=Azure IoT, O=MSFT, C=US\",\n" +
+            "                       \"notBeforeUtc\": \"2017-11-14T12:34:18Z\",\n" +
+            "                       \"notAfterUtc\": \"2017-11-20T12:34:18Z\",\n" +
+            "                       \"serialNumber\": \"000000000000000000\",\n" +
+            "                       \"version\": 3\n" +
+            "                   }\n" +
+            "               }\n" +
+            "           }\n" +
+            "       }\n" +
+            "   },\n" +
+            "   \"iotHubHostName\":\"" + IOT_HUB_HOST_NAME + "\",\n" +
+            "   \"deviceId\":\"" + DEVIECE_ID + "\",\n" +
+            "   \"initialTwinState\":{\n" +
+            "       \"tags\":{\n" +
+            "           \"tag1\":\"val1\",\n" +
+            "       },\n" +
+            "   },\n" +
+            "   \"provisioningStatus\":\"" + PROVISIONING_STATUS + "\",\n" +
+            "   \"createdDateTimeUtc\": \"" + CREATE_DATETIME_UTC_STRING + "\",\n" +
+            "   \"lastUpdatedDateTimeUtc\": \"" + LAST_UPDATED_DATETIME_UTC_STRING + "\",\n" +
+            "   \"etag\": \"" + ETAG + "\"\n" +
+            "}";
+
+            // act - assert
+            TestAssert.Throws<ProvisioningServiceClientException>(() => Newtonsoft.Json.JsonConvert.DeserializeObject<IndividualEnrollment>(invalidJson));
         }
 
-        /* SRS_DEVICE_ENROLLMENT_21_002: [The setRegistrationId shall throws ArgumentException if the provided registrationId is null, empty, or invalid.] */
+        [TestMethod]
+        [TestCategory("DevService")]
+        public void IndividualEnrollment_ConstructorJSON_ThrowsOnNonAttestation()
+        {
+            // arrange
+            string invalidJson =
+            "{\n" +
+            "   \"registrationId\":\"" + REGISTRATION_ID + "\",\n" +
+            "   \"iotHubHostName\":\"" + IOT_HUB_HOST_NAME + "\",\n" +
+            "   \"deviceId\":\"" + DEVIECE_ID + "\",\n" +
+            "   \"initialTwinState\":{\n" +
+            "       \"tags\":{\n" +
+            "           \"tag1\":\"val1\",\n" +
+            "       },\n" +
+            "   },\n" +
+            "   \"provisioningStatus\":\"" + PROVISIONING_STATUS + "\",\n" +
+            "   \"createdDateTimeUtc\": \"" + CREATE_DATETIME_UTC_STRING + "\",\n" +
+            "   \"lastUpdatedDateTimeUtc\": \"" + LAST_UPDATED_DATETIME_UTC_STRING + "\",\n" +
+            "   \"etag\": \"" + ETAG + "\"\n" +
+            "}";
+
+            // act - assert
+            TestAssert.Throws<ProvisioningServiceClientException>(() => Newtonsoft.Json.JsonConvert.DeserializeObject<IndividualEnrollment>(invalidJson));
+        }
+
+        [TestMethod]
+        [TestCategory("DevService")]
+        public void IndividualEnrollment_ConstructorJSON_ThrowsOnNonEtag()
+        {
+            // arrange
+            string invalidJson =
```

Comment:
Nit: `invalidJsonMissingEtag`. Got a bit confused reading the code. Name of the test is clarifying though.

Reply:
Done